Assignment/palindrome_test.c

```diff
@@ -0,0 +1,123 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// Palindrome_test.c
+// Program to test the decimal conversion.
+// ------------------------------------------------------------------------------------------------
+#pragma warning (disable:4996)
+#include "Palindrome.h"
+#include <stdio.h>
+#include <string.h> 
+#include <limits.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <ctype.h>
+
+#define INVALID_INPUT "Invalid Input"
+// ANSI color code
+#define GREEN_TEXT "\033[0;32m"
+#define RED_TEXT "\033[0;31m"
+#define RESET_TEXT "\033[0m"
+
+int IsValid (const char* num);
+void TestCases ();
+void UserInput ();
+
+/// <summary>To check if the input number is valid</summary>
+int IsValid (const char* num) {
+   if (*num == '\0' || num == NULL) return 0;
+   if (*num == '-') num++;
+   while (*num) {
+      if (!isdigit ((unsigned char)*num)) return 0;
+      num++;
+   }
+   return 1;
+}
+
+void TestCases () {
+   char* inputs[] = { "vini", "I did,did I", "Ma*)d(*am", "@@@##!!!", "Mam!",
+                      "Was it a car or a cat I saw?", "Don't nod", "Hello",
+                      "4334", "2147447412", "34543", "-121", "*-23" ,"?@343", "6", " " },
+      * expOutputs[] = { "Not a Palindrome", "Palindrome", "Palindrome", "Invalid Input",
+                             "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome",
+                             "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome",
+                             "Invalid Input", "Invalid Input", "Palindrome", "Invalid Input" };
+   int numOfTestCases = sizeof (inputs) / sizeof (inputs[0]);
+   printf ("\n-------Testcases for String and Integer Input------\n\n"
+           "| %-30s | %-20s | %-20s | %-10s |\n"
+           "|--------------------------------|----------------------|"
+           "----------------------|------------------|\n",
+           "Input", "Expected Output", "Actual Output", "Test Case Result");
+   for (int i = 0; i < numOfTestCases; i++) {
+      int actualOutput, reversed = 0, num = atoi (inputs[i]);
+      char* input = inputs[i];
+      actualOutput = IsValid (input) ? ReverseNumber (num, &reversed),
+         (num == reversed) : PalindromeCheck (input);

```

Comment:
You should have 2 test methods, 1 to check palindrome, 1 for reversal. Right now you are only checking isPalindrome, and you are not testing reversal at all. 
```suggestion
      actualOutput = PalindromeCheck (input);

```